wlc/__init__.py

```diff
@@ -620,6 +620,7 @@ class Component(LazyObject, RepoObjectMixin):
         "source_language",
         "priority",
         "is_glossary",
+        "category",
```

Comment:
It should be added to list of optionals below, otherwise wlc would fail on older weblate. This should be covered by the tests as well.

Reply:
I've added the field to the list. Also I've added a new test case class, `ComponentCompatibilityTest` for a component response with no optional fields support test.